### PR TITLE
Replacing MATLAB Language bundle

### PR DESCRIPTION
**Pull request by gauravsavadi**
We would like to replace the MATLAB language bundle used by Linguist for syntax highlighting MATLAB code.

## Description
We would like to replace the language grammar used by Linguist for syntax highlighting MATLAB code (currently matlab.tmbundle) with the official MathWorks Inc. grammar found [here](https://github.com/mathworks/MATLAB-Language-grammar).

We believe our rules do improve MATLAB syntax highlighting on GitHub and hope you switch over to using our rules in place of the existing TextMate bundle. Furthermore, because MathWorks develops the MATLAB language, we are committed to maintaining the accuracy of its syntax highlighting as the language evolves.

Please find more details about this request in [this issue](https://github.com/github/linguist/issues/3904).

## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a syntax highlighting grammar.
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x] **I am changing the source of a syntax highlighting grammar**
  - Old: [TextMate grammar results](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fgithub.com%2Ftextmate%2Fmatlab.tmbundle%2Fblob%2Fmaster%2FSyntaxes%2FM.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fmathworks%2FMATLAB-Language-grammar%2Fblob%2Fmaster%2FCircleArea.m&code=)
  - New: [MathWorks grammar results](https://github-lightshow.herokuapp.com/?utf8=✓&scope=from-url&grammar_url=https%3A%2F%2Fgithub.com%2Fmathworks%2FMATLAB-Language-grammar%2Fblob%2Fmaster%2FMatlab.tmbundle%2FSyntaxes%2FMATLAB.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fmathworks%2FMATLAB-Language-grammar%2Fblob%2Fmaster%2FCircleArea.m&code=)

As you can see, the TextMate bundle incorrectly flags the double quoted strings as errors. In addition, the file also contains an example of a command dual (axis equal) which the TextMate bundle does not recognize. Our grammar correctly recognizes these features and highlights them.

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

**Comment by pchaigno** [comment]
> As you can see, the TextMate bundle incorrectly flags the double quoted strings as errors. In addition, the file also contains an example of a command dual (axis equal) which the TextMate bundle does not recognize. Our grammar correctly recognizes these features and highlights them.

:+1: Thanks for the additional details!

@gauravsavadi Looks good to me. Could you please update your branch? There are some conflicts with our master branch.

**Comment by gauravsavadi** [comment]
@pchaigno 

The conflict has been resolved.

**Comment by lildude** [comment]
Thanks for the PR @gauravsavadi and good to see you've resolved the issues you encountered before.

The current failures are due to the `grammar.yml` file not being automatically updated to include info about your new grammar for some reason. This should have happened when you ran `script/add-grammar --replace...`.

Did you definitely run this command to add your grammar?

I've just run this locally and it worked for me... almost. It left an empty `vendor/grammars/matlab.tmbundle` element, but it did correctly add the new section:

```diff
diff --git a/grammars.yml b/grammars.yml
index 38e1343e..e6892f1d 100755
--- a/grammars.yml
+++ b/grammars.yml
@@ -48,6 +48,8 @@ vendor/grammars/Lean.tmbundle:
 - source.lean
 vendor/grammars/LiveScript.tmbundle:
 - source.livescript
+vendor/grammars/MATLAB-Language-grammar:   <---- This is what you're missing.
+- source.matlab
 vendor/grammars/MQL5-sublime:
 - source.mql5
 vendor/grammars/MagicPython:
@@ -541,9 +543,7 @@ vendor/grammars/marko-tmbundle:
 - text.marko
 vendor/grammars/mathematica-tmbundle:
 - source.mathematica
-vendor/grammars/matlab.tmbundle:
-- source.matlab
-- source.octave
+vendor/grammars/matlab.tmbundle: []   <--- This should have been removed completely.
 vendor/grammars/maven.tmbundle:
 - text.xml.pom
 vendor/grammars/mediawiki.tmbundle:
```

To resolve this, you'll need to manually modify the `grammars.yml` file to remove the old `vendor/grammars/matlab.tmbundle` section (don't add the line shown above) and add the `vendor/grammars/MATLAB-Language-grammar` section shown above.

/cc @vmg - looks like the recent change to the grammar compiler is leaving empty elements in the `grammars.yml` file for removed/replaced grammars.

**Comment by vmg** [comment]
Fixed in https://github.com/github/linguist/pull/4001 :)

**Comment by lildude** [comment]
Whoops, we didn't mean for this to be closed. Re-opening.

**Comment by gauravsavadi** [comment]
@lildude 

I definitely ran the add-grammar command with the syntax:

`.script/add-grammar --replace matlab.tmbundle https://github.com/mathworks/MATLAB-Language-grammar`

I've updated the grammars.yml file as you suggested.

**Comment by lildude** [comment]
> I definitely ran the add-grammar command with the syntax:

Hmmm. Not sure what went wrong then. Will keep an eye on things in case this happens again.

In the mean time, thanks for the update and PR. This is looking ✨ and the highlighting is a clear improvement.

I'll hold off my final approval until @pchaigno or @Alhadis have had a chance to approve.

I'll make a new release once this PR has been merged.

File: grammars.yml [hunk 1]
```diff
@@ -541,9 +543,6 @@ vendor/grammars/marko-tmbundle:
 - text.marko
 vendor/grammars/mathematica-tmbundle:
 - source.mathematica
-vendor/grammars/matlab.tmbundle:
-- source.matlab
-- source.octave
```
Comment (Alhadis) [inline grammars.yml]: Some grammars might be embedding the `source.octave` scope, so I recommend adding a copy of `source.matlab` which aliases this. Even something as simple as a wrapper which embeds `source.matlab` using the `source.octave` scope.

~~~xml
<dict>
		<key>include</key>
		<string>source.octave</string>
</dict>
~~~

Comment (Alhadis) [reply grammars.yml]: Having said that, I haven't checked to see if there *are* grammars which use this scope, or whether Octave and MATLAB have any noticeable differences (I actually know nothing about either, save that Octave is a FOSS alternative).

YMMV.

**Comment by lildude** [comment]
Woohoo!! Last PR for the release. Hope to have it done and deployed by EoD :gb: time today.

**Comment by gauravsavadi** [comment]
@lildude 

Thanks for merging the changes, and great to know that this is live now. 

I just had one clarifying question.
The coloring on GitHub seems a little different from the [Lightshow](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fgithub.com%2Fmathworks%2FMATLAB-Language-grammar%2Fblob%2Fmaster%2FMatlab.tmbundle%2FSyntaxes%2FMATLAB.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fmathworks%2FMATLAB-Language-grammar%2Fblob%2Fmaster%2FCircleArea.m&code=) page.

For example, compare the Lightshow result to the actual file [here](https://github.com/mathworks/MATLAB-Language-grammar/blob/master/CircleArea.m). 

The coloring is different. Could you clarify on why this is?

**Comment by lildude** [comment]
> The coloring is different. Could you clarify on why this is?

Yup... Lightshow is still using the old source for the GitHub site colours and not the new [Primer](https://github.com/primer/primer) based colours. 

I've completely missed that the pkg name is the same but the source isn't. I'll update it :soon:

**Comment by lildude** [comment]
> I'll update it 🔜

Done. They look the same now. Thanks for pointing it out.

**Pull request merged**